### PR TITLE
Ckedito5 patch for table cell resize

### PR DESCRIPTION
**Pull request by sharmasahil**
### Jira

[SD-358] - Control WYSIWYG and data table column widths in ckeditor

### Fix

1. Removed local patch as it got issues with retaining the styles while we edit the node again, we are already using Limit allowed html filter which is causing it.
2. Found this sandbox module https://git.drupalcode.org/sandbox/s_leu-3413350/-/tree/1.0.x which solves this issue and allows table inline styles to be saved correctly. It provides a new filter in text format 'Resize table columns' enabling it sorts col resize and retain styles on edit as well.
3. Updated basic enhancer to provide table and col inline styles to FE.

[SD-358]: https://digital-vic.atlassian.net/browse/SD-358?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by vincent-gao — CHANGES_REQUESTED** [review]
hey @sharmasahil 
do you need an update hook to apply the changes across all sites?
Also, could you clarify the need for a local patch?

**Comment by sharmasahil** [comment]
> hey @sharmasahil do you need an update hook to apply the changes across all sites? Also, could you clarify the need for a local patch?

Thanks for the review @vincent-gao, I have updated the PR with comments and also removed local patch, put install hook too.

File: modules/tide_landing_page/src/Plugin/jsonapi/FieldEnhancer/BasicTextEnhancer.php [hunk 1]
```diff
@@ -82,4 +83,73 @@ public function replaceUnicodeWhitespace(array $data) {
...
+                $processedCol->setAttribute($dataAttribute, $value);
+              }
+            }
+          }
+        }
+      }
+
+      // Extract only the inner HTML of the <body> tag, avoiding extra tags.
+      $processedBodyContent = '';
+      foreach ($processedDom->getElementsByTagName('body')->item(0)->childNodes as $node) {
```
Comment (vincent-gao) [inline modules/tide_landing_page/src/Plugin/jsonapi/FieldEnhancer/BasicTextEnhancer.php:145]: can this be simplified by using `DOMDocument::saveHTML` directly on the <body> element.
e.g.  
```php
$processedBodyContent = $processedDom->saveHTML($processedDom->getElementsByTagName('body')->item(0));
```

File: modules/tide_landing_page/src/Plugin/jsonapi/FieldEnhancer/BasicTextEnhancer.php [hunk 2]
```diff
@@ -82,4 +83,73 @@ public function replaceUnicodeWhitespace(array $data) {
...
+    if (isset($data['value']) && isset($data['processed'])) {
+      // Load 'value' and 'processed' HTML as DOM objects for manipulation.
+      $valueDom = new \DOMDocument();
+      $processedDom = new \DOMDocument();
+
+      // Suppress warnings for malformed HTML in $value and $processed.
+      libxml_use_internal_errors(true);
+      $valueDom->loadHTML($data['value']);
+      $processedDom->loadHTML($data['processed']);
+      libxml_clear_errors();
```
Comment (vincent-gao) [inline modules/tide_landing_page/src/Plugin/jsonapi/FieldEnhancer/BasicTextEnhancer.php:97]: 👍

File: modules/tide_ckeditor/modules/ckeditor_tablecol_resize/js/ckeditor5_plugins/tableColResize/src/tablecolresizeediting.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+			} )
+		);
+
+    // Ensures that the value of the data-resize-width attribute is added to the
+    // ckeditor model when the editor loads.
+    editor.conversion.for( 'upcast' ).add( dispatcher =>
+      dispatcher.on( `element:col`, ( evt, data, conversionApi ) => {
+        const { schema, writer } = conversionApi;
+        const colWidth = data.viewItem.getAttribute( dataAttribute );
+        console.log('col upcast', colWidth);
```
Comment (vincent-gao) [inline modules/tide_ckeditor/modules/ckeditor_tablecol_resize/js/ckeditor5_plugins/tableColResize/src/tablecolresizeediting.js:54]: the console.log statement should be removed or replaced with a proper logging mechanism if needed for debugging.

**Review by vincent-gao — APPROVED** [review]
hey @sharmasahil 
Just a few minor things, and it's perfectly fine if you don't want to make any changes.

**Pull request merged**